### DISCUSSION TITLE
Support gifs in gatsby-image

### DISCUSSION THREAD

**Discussion by KyleAMathews** [gatsbyjs/gatsby#29410] [Ideas / Feature Requests]
People often complain that gatsby-image doesn't support gifs — and end up doing [weird work arounds](https://github.com/wesbos/wesbos/blob/master/src/components/Img.js) to make it work:

<img width="790" alt="Screen Shot 2020-05-01 at 10 34 06 AM" src="https://user-images.githubusercontent.com/71047/80822245-2248c500-8b8f-11ea-9ea5-f0523806e84b.png">

We could just [directly support gifs in gatsby-image like @wesbos' component](https://github.com/wesbos/wesbos/blob/master/src/components/Img.js) — that + better support for querying images in graphiql would alleviate a lot of complaints about querying images .

**Comment by bstrokevin** [comment]
The issue if the feature image is a gif, the feature image don't have pubicURL(workaround option) and childImageSharp is null. The feature image only have Source URL which is not local. Can any know how to fix this?

**Comment by ghost** [comment]
Gatsby Image is focused on performance. GIFs are not at all performant. They also don't support alpha transparency. If moving pictures are to be added to Gatsby image (which would make for an interesting API no doubt) please consider using performant BGP images which about 98% smaller than GIFs _and_ support alpha transparency. (Note: Browser support for BPG is lacking but Fabrice created a WASM-based codec to load them).

https://bellard.org/bpg/animation.html

**Comment by DSchau** [comment]
Thinking _very_ briefly about this -- I think what we'd want to do is:

Enable querying by a GIF, however this would be wrapped by a performant / optimized [`video` element](https://caniuse.com/#feat=video).

The specifics would be the interesting piece (e.g. what tools do we use, what is the API, etc.) but I think this could be a fairly interesting use case, particularly if similar functionality is implemented and available in the Remark / MDX space.

**Comment by timhagn** [comment]
Hi there,

some time ago I wrote a plugin & transformer for gifs, though never found the time to make it a fully functional package (README, etc.).
Should I publish it nevertheless and we'll see how to go from there?
Have a look at https://github.com/timhagn/giffit : ).

Best,

Tim.
